tests/test_l10n_download_links.py

```diff
@@ -7,74 +6,47 @@
 import pytest
 import requests
 
+pytestmark = [pytest.mark.nondestructive, pytest.mark.skip_selenium]
 
-nondestructive = pytest.mark.nondestructive
-skip_selenium = pytest.mark.skip_selenium
 
+def pytest_generate_tests(metafunc):
+    base_url = metafunc.config.option.base_url
+    page = metafunc.cls.page  # get the target page from the test class
+    r = requests.get('{0}{1}'.format(base_url, page))
+    soup = BeautifulSoup(r.content, 'html.parser')
+    table = soup.find('table', class_='build-table')  # find the table of links
+    idlist = []
+    argvalues = []
+    for row in table.find('tbody').find_all('tr'):
+        locale = row['id']
+        idlist.append(locale)  # the test id should be the locale
+        url = row.find('a')['href']
+        argvalues.append([locale, url])  # pass locale and url to each test
+    metafunc.parametrize('locale, url', argvalues, ids=idlist)
 
-@nondestructive
-@skip_selenium
-class TestLocalisedDownloadLinks:
 
-    def get_language_rows(self, mozwebqa, link='/firefox/all/'):
-        '''
-            Get all links for each locale on the page under test.
-            The default page under test is /firefox/all/ .
-        '''
-        url = "%s/%s" % (mozwebqa.base_url, link)
-        page_response = requests.get(url)
-        html = BeautifulSoup(page_response.content)
-        language_rows = html.findAll('tr', id=True)
-        return language_rows
+def check_link(locale, url):
+    r = requests.head(url, allow_redirects=True)
+    assert locale in r.url  # the correct locale is in the final url
+    assert requests.codes.found == r.history[0].status_code  # url redirects
```

Comment:
The old code allows for both 301 and 302. Is this new code okay, or should we be allowing for both of those still?